### DISCUSSION TITLE
No logs appearing for an Airflow DAG

### DISCUSSION THREAD

**Discussion by ariansacct** [apache/airflow#36643] [Q&A]
I have created a workflow in `dags` folder. The workflow appears in the Airflow UI, but no logs exist, either under graph/tasks/log or in `~/airflow/logs` folder. I then manually created a `dag_id=<dag_name>` folder under `logs`, but still nothing. The workflow and its underlying tasks do appear. Logs for the example DAGs show up fine. Tried with Airflow 2.8, Python 3.8.2 on macOS Sonoma 14.0, as well as Airflow 2.7.3, Python 3.8.10 on Ubuntu 20.04.6 LTS.
Please let me know what more information is needed.

**Comment by potiuk** [comment]
What logs do you expect? UI will not produce any logs for dags. There are logs in dag file processor/scheduler when they are parsed, and task logs when they are executed. Maybe you look at wrong machine (where webserver is running) and not where your scheduler and workers are running.

**Comment by potiuk** [comment]
Or maybe you have not run your dags but expect the logs from running them ?

**Reply by ariansacct** [reply]
I do run the DAG (the play button to its right -> Trigger DAG). It runs successfully and finishes with no output. Its runs are recorded correctly.
I have all these in my code, in this order:

```
import logging
logging.debug('This is a debug message')
logging.info('This is an info message')
logging.warning('This is a warning message')
logging.error('This is an error message')
logging.critical('This is a critical message')
```
and
```
logger = logging.getLogger("airflow.task")
logger.error("Your custom error")
```
and
```
print('This is a print')
```

None are getting printed. In addition, there are no folders for this particular DAG under `logs/`, unlike for others. The folder is the right one mentioned in `airflow.cfg`.

All the Airflow services are on the same machine.

**Reply by potiuk** [reply]
If it's during parsing the dag - look in dag file processor log, if it's during execution - look at execution. You have not explained what your deployment is so it's hard to help you without information  how you deployed your dags, where your tasks are running etc.

It's likely you are simply looking in a completely different environment than where your Airlfow is running.

But what your environment is, only you know and  if you explain it, then there is a chance someone will be able to help. But only after you ecplain it. There are many ways of installing and deploying airflow https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html and I suggest you explain your deployment following the terminology and types of deployment explained there if you want to get help.
